tests/unittests/utils/test_qwen3_vl.py

```
@@ -670,5 +671,118 @@ def test_embed_prompts_empty_raises(self):
+    def test_embed_frames_empty_raises(self):
+        model = self._make_model()
+        with pytest.raises(ValueError, match="empty"):
+            model.embed_frames([])
+
+    def test_embed_frames_subsamples_to_video_fps(self):
+        model = self._make_model()
+        model.config.video_fps = 2.0
+
+        captured = {}
+
+        def _capture(messages):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_capture`

(ANN202)